packages/platform/src/i18n.ts

```diff
@@ -52,16 +52,21 @@ async function loadTranslationsForComponent (component: Component): Promise<Mess
 }
 
 async function getTranslation (message: IntlString): Promise<IntlString | Status> {
-  const id = parseId(message)
-  let messages = translations.get(id.component)
-  if (messages === undefined) {
-    messages = await loadTranslationsForComponent(id.component)
-    translations.set(id.component, messages)
-  }
-  if (messages instanceof Status) {
-    return messages
+  try {
+    const id = parseId(message)
+    let messages = translations.get(id.component)
+    if (messages === undefined) {
+      messages = await loadTranslationsForComponent(id.component)
+      translations.set(id.component, messages)
+    }
+    if (messages instanceof Status) {
+      return messages
+    }
+    return (id.kind ? (messages[id.kind] as Record<string, IntlString>)?.[id.name] : messages[id.name] as IntlString) ?? message
+  } catch (err) {
+    setPlatformStatus(err)
+    return err
```

Comment:
This is not what we want, we want original string to be returned (but only in case when parseId failed). I propose to get back to your original version, ideally introducing `isValidId`.